components/formats-bsd/src/loci/formats/in/OBFReader.java

```diff
@@ -152,9 +156,28 @@ protected void initFile(String id) throws FormatException, IOException
     final int lengthOfDescription = in.readInt();
     final String description = in.readString(lengthOfDescription);
     metadata.put("Description", description);
+    String ome_xml = "" ;
     if (fileVersion > 1)
     {
-      in.readLong();
+      final long meta_data_position = in.readLong() ;
+      final long current_position = in.getFilePointer() ;
+      
+      in.seek( meta_data_position ) ;
+
+      for (String key = readString() ; key.length() > 0 ; key = readString())
+      {
+        if (key.equals( "ome_xml" ))
+        {
+          ome_xml = readString() ;
+          break ;
+        }
+        else
+        {
+          readString() ;
```

Comment:
Are keys other than ```ome_xml``` likely to occur?  If so, it might be useful to store those key/value pairs in the original metadata table, by calling ```addGlobalMeta(key, readString())``` here.